action.yml

```diff
@@ -0,0 +1,37 @@
+name: Setup Go Tooling
+description: Setup Go tooling including cached Go modules.
+
+inputs:
+  cacheKey:
+    description: Custom cache key used
+    required: false
+    default: ci
+
+runs:
+  using: composite
+
+  steps:
+    - name: Set up Go
+      uses: actions/setup-go@v2.1.5
+      with:
+        go-version: ^1.17
+
+    - name: Cache Go modules
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.cache/go-build
+          ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ inputs.cacheKey }}-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-${{ inputs.cacheKey }}-
+
+    - name: Cache sage folders
+      uses: actions/cache@v2
+      with:
+        path: |
+          ./.sage/tools
+          ./.sage/bin
+        key: ${{ runner.os }}-sage-${{ inputs.cacheKey }}-${{ hashFiles('./.mage/go.sum') }}
```

Comment:
should this be `./.sage/`?

Reply:
We only want to cache generated files. We don't want to cache the go files themselves :)